sklearn/mixture/_base.py

```diff
@@ -119,16 +119,18 @@ def _initialize_parameters(self, X, random_state):
             )
             resp[np.arange(n_samples), label] = 1
         elif self.init_params == "random":
-            resp = random_state.uniform(size=(n_samples, self.n_components))
+            resp = random_state.uniform(size=(n_samples, self.n_components)).astype(
+                X.dtype
+            )
```

Comment:
I am not sure this makes a big difference but we may as well avoid a copy if `X.dtype` is `float64` (which I guess is the most common use case)?
```suggestion
            resp = np.asarray(
                random_state.uniform(size=(n_samples, self.n_components)), dtype=X.dtype
            )
```